realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1525,6 +1525,15 @@ public void executeTransaction(Transaction transaction) {
             throw new IllegalArgumentException("Transaction should not be null");
         }
 
+        // Warn on transaction being executed on UI thread if allowWritesOnUiThread is set to true, throw otherwise
+        if (sharedRealm.capabilities.isMainThread()) {
+            if (getConfiguration().isAllowWritesOnUiThread()) {
+                RealmLog.warn("It is not recommended to run transactions on the UI thread as it may lead to a drop of frames or ANRs. Please consider using 'executeTransactionAsync' instead.");
```

Comment:
Not sure we should log this as a warning if enabled. It is only enabled if you manually opt into it, so hopefully, people are already aware of the tradeoff and the warn level will most likely become visible to users since that is the default setting.

Reply:
What should we do then? User `RealmLog.info` instead?

Reply:
I would probably just remove it completely